tun/tcp_offload_linux.go

```diff
@@ -189,14 +189,29 @@ func tcpPacketsCanCoalesce(pkt []byte, iphLen, tcphLen uint8, seq uint32, pshSet
 			return coalesceUnavailable
 		}
 	}
-	if pkt[1] != pktTarget[1] {
-		// cannot coalesce with unequal ToS values
-		return coalesceUnavailable
-	}
-	if pkt[6]>>5 != pktTarget[6]>>5 {
-		// cannot coalesce with unequal DF or reserved bits. MF is checked
-		// further up the stack.
-		return coalesceUnavailable
+	if pkt[0]>>4 == 6 {
+		if pkt[0] != pktTarget[0] || pkt[1]>>4 != pktTarget[1]>>4 {
+			// cannot coalesce with unequal Traffic class values
+			return coalesceUnavailable
+		}
+		if pkt[7] != pktTarget[7] {
+			// cannot coalesce with unequal Hop limit values
+			return coalesceUnavailable
+		}
+	} else {
+		if pkt[1] != pktTarget[1] {
+			// cannot coalesce with unequal ToS values
+			return coalesceUnavailable
+		}
+		if pkt[6]>>5 != pktTarget[6]>>5 {
```

Comment:
I think this includes the More Fragments bit, which is supposed to differ in the last fragment of the frame.

I note that fragments are kicked out of the fastpath later in this code, so it may not matter what happens here.

Reply:
Yes it does. MF is checked further up in `tcpGRO()`, so this is unreachable if MF is set.